### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by JeffreyNix**
Finishing up Put test but everything else should be there

### CODE REVIEW

**Comment by CDobbins1** [comment]
Reviewing this

File: SecretSanta/src/SecretSanta.Api/Controllers/IController.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SecretSanta.Business.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SecretSanta.Api.Controllers
+{
+    public interface IController<TEntity>
```
[the remark this answers is not in the document]
Comment (JeffreyNix) [reply SecretSanta/src/SecretSanta.Api/Controllers/IController.cs]: yeah, I really should migrate things over to that. it was a quick hack to try and make things work generically for unit tests and I never got back to it.

File: SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            return NotFound();
+        }
+
+        // DELETE: api/ApiWithActions/5
+        [HttpDelete("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<bool>> Delete(int id)
+        {
+            if (await GiftService.FetchByIdAsync(id) is { } gift)
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs]: Local variable 'gift' is never used.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 3]
```diff
@@ -12,9 +18,26 @@ public class Startup
     {
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
-        //public void ConfigureServices(IServiceCollection services)
-        //{
-        //}
+        public static void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<ApplicationDbContext>();
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Though this technically registers the `ApplicationDbContext` type, it does not actually configure the db context and will fail on all DB interaction.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 4]
```diff
@@ -12,9 +18,26 @@ public class Startup
...
+
+            services.AddScoped<IGroupService, GroupService>();
+            services.AddScoped<IUserService, UserService>();
+            services.AddScoped<IGiftService, GiftService>();
+
+            services.AddScoped<UserController, UserController>();
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: You should not need to register the classes directly. All usages [should depend on the interface abstraction](https://deviq.com/dependency-inversion-principle/) and not the concrete type.

**Pull request closed without merging**